src/bda/plone/orders/browser/bookings.py

```diff
@@ -108,47 +115,109 @@ class BookingsTable(BrowserView):
     table_id = 'bdaplonebookings'
     data_view_name = '@@bookingsdata'
 
-    def render_group_filter(self):
+    def render_filter(self):
         # group orders
         groups = vocabs.groups_vocab()
         group_selector = factory(
-            'label:select',
+            'div:label:select',
             name='group',
             value=self.request.form.get('group', ''),
             props={
+                'div.class': 'group_filter',
                 'vocabulary': groups,
                 'label': _('group_orders_by',
                            default=u'Group orders by'),
             }
         )
-        filter = group_selector(request=self.request)
-        return filter
+        filter_widgets = group_selector(request=self.request)
 
-    def render_date_from_filter(self):
+        # From date filter.
         from_date = factory(
-            'label:text',
+            'div:label:text',
             name='from_date',
             value=self.request.form.get('from_date', ''),
             props={
+                'div.class': 'date_from_filter',
                 'label': _('from_date',
                            default=u'Filter from date'),
             }
         )
-        filter = from_date(request=self.request)
-        return filter
+        filter_widgets += from_date(request=self.request)
 
-    def render_date_to_filter(self):
+        # To date filter.
         to_date = factory(
-            'label:text',
+            'div:label:text',
             name='to_date',
             value=self.request.form.get('to_date', ''),
             props={
+                'div.class': 'date_to_filter',
                 'label': _('to_date',
                            default=u'to date'),
             }
         )
-        filter = to_date(request=self.request)
-        return filter
+        filter_widgets += to_date(request=self.request)
+
+        # vendor areas of current user
+        vendors = vendors_form_vocab()
+        vendor_selector = None
+        # vendor selection, include if more than one vendor
+        if len(vendors) > 2:
```

Comment:
We might should skip vendor filter for "bookings in context" and "orders in context" or reduce the vocabulary to all vendors found in the specific context if we want to keep the filter.